src/dash-table/derived/table/index.tsx

```diff
@@ -26,25 +28,53 @@ const handleSetFilter = (
     setState({workFilter: {map, value: filter_query}, rawFilterQuery});
 };
 
-function propsAndMapFn(propsFn: () => ControlledTableProps, setFilter: any) {
+function propsAndMapFn(
+    propsFn: () => ControlledTableProps,
+    setFilter: any,
+    toggleFilterOptions: (column: IColumn) => IColumn
+) {
     const props = propsFn();
 
-    return R.merge(props, {map: props.workFilter.map, setFilter});
+    return R.merge(props, {
+        map: props.workFilter.map,
+        setFilter,
+        toggleFilterOptions
+    });
 }
 
 export default (propsFn: () => ControlledTableProps) => {
     const setFilter = memoizeOne((setProps: SetProps, setState: SetState) =>
         handleSetFilter.bind(undefined, setProps, setState)
     );
 
+    const toggleFilterOptions = memoizeOne(
+        (setProps: SetProps, columns: IColumn[]) => (column: IColumn) => {
```

Comment:
Called by https://github.com/plotly/dash-table/pull/893/files#r622661715

Defined at the factory level so that the individual components remain unaware of `columns` for memoization/cache busting / performance reasons and `setProps` so they remain unaware of Dash's idiosyncracies.